push-sender/src/main/java/org/jboss/aerogear/unifiedpush/message/cache/SimpleApnsClientCache.java

```diff
@@ -101,12 +101,17 @@ public void disconnectOnChange(@Observes final iOSVariantUpdateEvent iOSVariantU
         }
     }
 
+    public void disconnectOnChangeEvent(@Observes final iOSVariantUpdateEvent iOSVariantUpdateEvent) {
+        final iOSVariant variant = iOSVariantUpdateEvent.getiOSVariant();
+        disconnectOnChange(variant);
+    }
+
     private String extractConnectionKey(final iOSVariant iOSVariant) {
         final StringBuilder sb = new StringBuilder()
                 .append(iOSVariant.getVariantID())
```

Comment:
I know this is not directly part of this PR, but this could be changed to `new StringBuilder(iOSVariant.getVariantID())` to avoid one useless `append` (the `StringBuilder()` constructor create StringBuilder with capacity 16, that we immediately expand with the next append)